rest/rest/src/main/java/io/syndesis/rest/v1/handler/connection/ConnectorHandler.java

```diff
@@ -141,4 +153,44 @@ public Kind resourceKind() {
         final Map<String, String> props) {
         return verifier.verify(connectorId, encryptionComponent.decrypt(props));
     }
+
+    @GET
+    @Path("/{id}/icon")
+    public Response getConnectorIcon(@NotNull @PathParam("id") final String connectorId) {
+        Connector connector = get(connectorId);
+        String connectorIcon = connector.getIcon();
+
+        // If there is no specified icon, return 404
+        if (connectorIcon == null) {
+            return Response.status(Response.Status.NOT_FOUND).build();
+        }
+
+        // If the specified icon is a data URL, or a non-URL like value (e.g. font awesome class name), return 404
+        if (connectorIcon.startsWith("data:") || !connectorIcon.contains("/")) {
+            return Response.status(Response.Status.NOT_FOUND).build();
+        }
+
+        OkHttpClient httpClient = new OkHttpClient();
+        try {
+            final okhttp3.Response externalResponse = httpClient.newCall(new Request.Builder().get().url(connectorIcon).build()).execute();
+            String contentType = externalResponse.header(CONTENT_TYPE, MediaType.APPLICATION_OCTET_STREAM);
+            String contentLength = externalResponse.header(CONTENT_LENGTH);
+
+            StreamingOutput streamingOutput = (out) -> {
+                BufferedSink sink = Okio.buffer(Okio.sink(out));
+                sink.writeAll(externalResponse.body().source());
+                sink.close();
+            };
+
+            Response.ResponseBuilder actualResponse = Response.ok(streamingOutput, contentType);
+            if (!Strings.isEmpty(contentLength)) {
+                actualResponse.header(CONTENT_LENGTH, contentLength);
+            }
+
+            return actualResponse.build();
+        } catch (IOException e) {
+            throw new SyndesisServerException(e);
```

Comment:
Do we really want it to be a full-blown error when the icon can't be served from an external server which can go down anytime? I think that's too much of a dependency here. Instead, I would return a default icon.

Reply:
Good idea. A default placeholder icon. @syndesisio/uxd Could you please provide/suggest a default icon we can use for connectors?